pkgs/development/python-modules/triton/bin.nix

```diff
@@ -51,29 +51,8 @@ buildPythonPackage rec {
   # If this breaks, consider replacing with "${cuda_nvcc}/bin/ptxas"
   postFixup =
     ''
-      chmod +x "$out/${python.sitePackages}/triton/third_party/cuda/bin/ptxas"
-    ''
-    + (
-      let
-        # Bash was getting weird without linting,
-        # but basically upstream contains [cc, ..., "-lcuda", ...]
-        # and we replace it with [..., "-lcuda", "-L/run/opengl-driver/lib", "-L$stubs", ...]
-        old = [ "-lcuda" ];
-        new = [
-          "-lcuda"
-          "-L${addDriverRunpath.driverLink}"
-          "-L${cudaPackages.cuda_cudart}/lib/stubs/"
-        ];
```

Comment:
I doubt it's correct to remove it (hence [wip]). I expect the fix is to add the directories here: https://github.com/triton-lang/triton/blob/main/third_party/nvidia/backend/driver.py#L20, but I'm not sure how to repro the original problem this code is trying to fix.

Reply:
> how to repro

Remove the patch, try running a gpu test in a nixos environment, observe that triton tries running ldconfig which doesn't exist

> I doubt it's correct to remove it

Yes please if you're using triton-bin do rewrite the patch. The intent is that in a nixos environment triton should load libcuda from `${addDriverRunpath.driverLInk}/lib` and not try to run ldconfig. It's ok (maybe even desirable) to keep the fallback path for non-nixos environments like ubuntu

Reply:
Thanks, I've pushed a fix. This one bypasses the fallback logic, since otherwise I'd have to start testing if the directories exist before invoking the fallback, which is a larger patch. I guess we could do that if we really care about falling back to ldconfig.

Personally, when I use this type of thing on non-nixos, I just set up `/run/opengl-driver/lib` to point to my driver install.

Reply:
> I just set up /run/opengl-driver/lib to point to my driver install.

That is definitely preferable to relying on ldconfig - from the user's PoV. On nixpkgs side though, I don't think we should be "breaking" upstream behaviour.